src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1528,1894 @@ interface MPSMatrixMultiplication {
+		void EncodeSequence (IMTLCommandBuffer commandBuffer, MPSMatrix[] sourceMatrices, MPSMatrix[] destinationMatrices, [NullAllowed] MPSRnnRecurrentMatrixState recurrentInputState, [NullAllowed] NSMutableArray<MPSRnnRecurrentMatrixState> recurrentOutputStates);
+
+		[Export ("encodeBidirectionalSequenceToCommandBuffer:sourceSequence:destinationForwardMatrices:destinationBackwardMatrices:")]
+		void EncodeBidirectionalSequence (IMTLCommandBuffer commandBuffer, MPSMatrix[] sourceSequence, MPSMatrix[] destinationForwardMatrices, [NullAllowed] MPSMatrix[] destinationBackwardMatrices);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("copyWithZone:device:")]
+		MPSRnnMatrixInferenceLayer Copy ([NullAllowed] NSZone zone, [NullAllowed] IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSNNImageNode")]
+	[DisableDefaultCtor]
+	interface MPSNnImageNode
+	{
+		[Export ("initWithHandle:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] IMPSHandle handle);
+
+		[Static]
+		[Export ("nodeWithHandle:")]
+		MPSNnImageNode FromHandle ([NullAllowed] IMPSHandle handle);
+
+		[Static]
+		[Export ("exportedNodeWithHandle:")]
+		MPSNnImageNode ExportedNodeWithHandle ([NullAllowed] IMPSHandle handle);
+
+		[NullAllowed, Export ("handle", ArgumentSemantic.Retain)]
+		IMPSHandle MPSHandle { get; set; }
+
+		[Export ("format", ArgumentSemantic.Assign)]
+		MPSImageFeatureChannelFormat Format { get; set; }
+
+		[Export ("imageAllocator", ArgumentSemantic.Retain)]
+		IMPSImageAllocator ImageAllocator { get; set; }
+
+		[Export ("exportFromGraph")]
+		bool ExportFromGraph { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSNNStateNode")]
+	[DisableDefaultCtor]
+	interface MPSNnStateNode
+	{
+		[NullAllowed, Export ("handle", ArgumentSemantic.Retain)]
+		IMPSHandle MPSHandle { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSNNFilterNode")]
+	[DisableDefaultCtor]
+	interface MPSNnFilterNode
+	{
+		[Export ("resultImage")]
+		MPSNnImageNode ResultImage { get; }
+
+		[NullAllowed, Export ("resultState")]
+		MPSNnStateNode ResultState { get; }
+
+		[NullAllowed, Export ("resultStates")]
+		MPSNnStateNode[] ResultStates { get; }
+
+		[Export ("paddingPolicy", ArgumentSemantic.Retain)]
+		IMPSNnPadding PaddingPolicy { get; set; }
+
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNConvolutionNode")]
+	interface MPSCnnConvolutionNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:")]
+		MPSCnnConvolutionNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights);
```

Comment:
@spouliot in the previous PR I left a question for you: https://github.com/xamarin/xamarin-macios/pull/2589/files#diff-d05bb29f8320ca898f34fbaeb9d0385dR2690

Reply:
yes, I prefer `Create*` since it has a verb - like all methods should :)